lib/trie.js

```diff
@@ -0,0 +1,66 @@
+module.exports = Trie;
+
+function Trie() {
+  this.methods = {};
+}
+
+//insert new nodes to Trie
+Trie.prototype.add = function(path, verb, handler) {
+  var node = this;
+  var parts = getPathParts(path);
+  if (parts.length) {
+    for (var i = 0; i < parts.length; i++) {
+      if (!(parts[i] in node)) {
+        node[parts[i]] = new Trie();
+      }
+      if (i + 1 === parts.length) {
+        addVerbHandler(node[parts[i]], verb, handler);
+      }
+      node = node[parts[i]];
+    }
+  } else { // if we would ever encounter this situation?
```

Comment:
@bajtos: yes. I purpose-fully put the comment for your views. 
The thing is, every handler, in our `rest-adapter.js` is related to one restClass, thus it will always have model name in the `fullPath`. So, was not sure if we should handle this case. 
TODO: Will remove the comment and add a unit-test